File: gui/src/domain/project.ts
export interface Project {
  id: string
  name: string
  description?: string
  sessionPrefix: string
  createdAt: string
  updatedAt: string
}

export interface ProjectWithCount extends Project {
  sessionCount: number
}

export interface CreateProjectRequest {
  name: string
  description?: string
}

export interface UpdateProjectRequest {
  name?: string
  description?: string
}

File: internal/domain/project.go
package domain

import "time"

// Project represents a collection of related sessions.
type Project struct {
	ID            string    `json:"id"`
	Name          string    `json:"name"`
	Description   string    `json:"description,omitempty"`
	SessionPrefix string    `json:"sessionPrefix"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

// ProjectMetadata is an alias for Project since all fields are metadata.
type ProjectMetadata = Project
